x-pack/plugins/infra/server/lib/adapters/nodes/lib/create_node_item.ts

```diff
@@ -4,18 +4,44 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { last } from 'lodash';
+import moment from 'moment';
 import { InfraNode, InfraNodeMetric } from '../../../../../common/graphql/types';
 import { InfraBucket, InfraNodeRequestOptions } from '../adapter_types';
+import { getBucketSizeInSeconds } from './get_bucket_size_in_seconds';
 
 // TODO: Break these function into seperate files and expand beyond just documnet count
 // In the code below it looks like overkill to split these three functions out
 // but in reality the create metrics functions will be different per node type.
+
+const findLastFullBucket = (
+  bucket: InfraBucket,
+  bucketSize: number,
+  options: InfraNodeRequestOptions
+) => {
+  const { buckets } = bucket.timeseries;
+  const to = moment.utc(options.timerange.to);
+  return buckets.reduce((current, item) => {
+    const date = moment.utc(item.key + bucketSize * 1000);
+    if (!date.isAfter(to) && item.doc_count > 0) {
+      return item;
+    }
+    return current;
+  }, last(buckets));
+};
+
 function createNodeMetrics(
   options: InfraNodeRequestOptions,
   node: InfraBucket,
   bucket: InfraBucket
 ): InfraNodeMetric[] {
-  return [{ name: 'count', value: bucket.doc_count }];
+  const { timerange, metrics } = options;
+  const bucketSize = getBucketSizeInSeconds(timerange.interval);
+  const lastBucket = findLastFullBucket(bucket, bucketSize, options);
+  return metrics.filter(metric => lastBucket[metric.type.toString()]).map(metric => {
```

Comment:
The only time I do stuff like this is because I'm trying to figure out the magic syntax to get the Typescript compiler to shut up. This mostly happens because the Typescript error are so cryptic. I find dealing with translating enums to string OR using it for a lookup to be a real pain and to be honest I'm not sure what the `idiomatic` way to do this in Typescript. This is something that is very natural in Javascript.

Reply:
So compilation fails for you without the `toString()`?

Reply:
Looks like it didn't need it after all... This code block was creating an indecipherable error at the time and so i took the shot gun approach to solving it... :D